drivers/spi/spi_dw.c

```diff
@@ -342,6 +342,12 @@ static int transceive(struct device *dev,
 
 	spi_context_lock(&spi->ctx, asynchronous, signal);
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	if (device_busy_check(dev) == 0) {
+	    device_busy_set(dev);
```

Comment:
And how is this going to work once the transaction is done? You need to call device_busy_clear() in the relevant place.